rtt/base/TaskCore.cpp

```diff
@@ -337,5 +352,12 @@ namespace RTT {
         }
     }
 
+    std::string TaskCore::getName() const {
+        return name;
+    }
+    void TaskCore::setName(const std::string& name) {
+        this->name = name;
+        this->cName = strdup(name.c_str());
```

Comment:
I have to admit that I jumped a bit to conclusion here ... I did see issues with the name being garbage in the trace and attributed it to c_str ... But I'm not so sure anymore.